client/ayon_core/plugins/publish/collect_farm_env_variables.py

```diff
@@ -0,0 +1,33 @@
+import os
+
+import pyblish.api
+
+from ayon_core.pipeline.publish import FARM_JOB_ENV_DATA_KEY
+
+
+class CollectCoreJobEnvVars(pyblish.api.ContextPlugin):
+    """Collect set of environment variables to submit with deadline jobs"""
+    order = pyblish.api.CollectorOrder - 0.45
+    label = "AYON core Farm Environment Variables"
+    targets = ["local"]
+
+    def process(self, context):
+        env = context.data.setdefault(FARM_JOB_ENV_DATA_KEY, {})
+        for key in [
+            # AYON
+            "AYON_BUNDLE_NAME",
+            "AYON_DEFAULT_SETTINGS_VARIANT",
+            "AYON_PROJECT_NAME",
+            "AYON_FOLDER_PATH",
+            "AYON_TASK_NAME",
+            "AYON_WORKDIR",
```

Comment:
I know it's technically not related to this PR - but _why_ do we need to pass the workdir? Isn't this workdir path also local to the current machine and not rootless? I have a feeling we should maybe, after this PR investigate this and potentially clean that up?

Reply:
It may be used here by the way: 
- https://github.com/ynput/ayon-core/blob/a08a1af331163492644fa8f5713328c95c51cc44/client/ayon_core/plugins/publish/collect_rendered_files.py#L176-L187

But I guess that should still be set by the "launch context" regardless?

Reply:
I did try to find out where it is used, and I didn't find anything. If you can create issue it would be nice.

Reply:
> It may be used here by the way:

Yes, but is it then used for anything?